server/methods/getUsersOfRoom.js

```diff
@@ -16,17 +16,32 @@ Meteor.methods({
 			throw new Meteor.Error('error-not-allowed', 'Not allowed', { method: 'getUsersOfRoom' });
 		}
 
-		const subscriptions = RocketChat.models.Subscriptions.findByRoomIdWhenUsernameExists(rid, { fields: { 'u._id': 1 } }).fetch();
-		const userIds = subscriptions.map((s) => s.u._id); // TODO: CACHE: expensive
-		const options = { fields: { username: 1, name: 1 } };
-
-		const users = showAll === true
-			? RocketChat.models.Users.findUsersWithUsernameByIds(userIds, options).fetch()
-			: RocketChat.models.Users.findUsersWithUsernameByIdsNotOffline(userIds, options).fetch();
+		const subscriptions = RocketChat.models.Subscriptions.findByRoomIdWhenUsernameExists(rid);
 
 		return {
-			total: userIds.length,
-			records: users,
+			total: subscriptions.count(),
+			records: await RocketChat.models.Subscriptions.model.rawCollection().aggregate([
+				{ $match: { rid } },
+
+				{
+					$lookup:
```

Comment:
mongodb applies the same [BSON Document Size](https://docs.mongodb.com/manual/reference/limits/#BSON-Document-Size) limit to lookup results.. it didn't reach this limit on my 200k users database, but I'm almost sure that limit will be hit on larger databases which will make this query to fail

Reply:
🙄

Reply:
@sampaiodiego are you sure about that? I made a fast search here:
https://docs.mongodb.com/manual/core/aggregation-pipeline-limits/#result-size-restrictions
and if I'm not wrong the limit just applies for results(16mb per item), I dont think your subscription + user will get that size, and we can use `$project` to reduce the size...could you check it again? <3

Reply:
*EACH* document in the result set is subject to the BSON Document Size limit, currently 16 megabytes;